quinn-proto/src/connection/streams.rs

```diff
@@ -1156,30 +1087,31 @@ impl Recv {
         }
     }
 
-    fn read_chunk(&mut self) -> Result<Option<Bytes>, ReadError> {
+    fn read_chunk(&mut self) -> StreamReadResult<Bytes> {
         match self.assembler.read_chunk()? {
             Some(bytes) => Ok(Some(bytes)),
             None => self.read_blocked().map(|()| None),
         }
     }
 
-    fn read_chunks(&mut self, chunks: &mut [Bytes]) -> Result<Option<usize>, ReadError> {
+    fn read_chunks(&mut self, chunks: &mut [Bytes]) -> Result<Option<(usize, usize)>, ReadError> {
```

Comment:
A comment to disambiguate which of the two returned `usize` is the number of chunks would help here.

Reply:
Yeah, this was a little opaque. I instead defined a simple struct `ReadChunks`, see what you think.